x-pack/plugins/monitoring/dev_docs/reference/terminology.md

```diff
@@ -6,4 +6,31 @@ The process of collecting monitoring data handled by the stack components themse
 
 #### Metricbeat collection
 
-The process of collecting monitoring data using metricbeat. Each component exposes an endpoint that metricbeat queries using a module for that component. Metricbeat then sends the data to elasticsearch for all monitored components. 
\ No newline at end of file
+The process of collecting monitoring data using metricbeat. Each component exposes an endpoint that metricbeat queries using a module for that component. Metricbeat then sends the data to elasticsearch for all monitored components.
+
+#### Production deployment
+
+The deployment that receives customer service data, as distinguished from [monitoring deployment](#monitoring-deployment). For example, given two deployments:
+
+- A: the deployment that stores the customer's service observability data
+- B: the deployment that stores the monitoring data
+
+The production deployment will be "A" and the "monitoring" deployment will be "B".
+
+If the deployment is [self-monitoring](#self-monitoring), then the production deployment is the same as the monitoring deployment.
+
+#### Monitoring deployment
+
+The deployment that receives stack monitoring data from the production deployment(s), as distinguished from the [production deployment](#production-deployment).
+
+#### Self-monitoring
+
+The practice of storing stack monitoring data in the same deployment as the customer service observability data.
+
+This is useful for testing and development, but not recommended for production.
+
+This is because you typically need the monitoring data when the production deployment is down or slow to ingest. If they're the same deployment, the monitoring data will be unavailable as well and useless for troubleshooting.
+
+#### Standalone Cluster
```

Comment:
I know this is the terminology we currently use, but does it make sense, I'm curious? I've always been thrown off by it because it seems like that's not really what we mean... is it supposed to mean "standalone" in the sense that it's actually NOT a cluster, it's a standalone product not attached to any cluster by way of cluster_uuid?

Reply:
For this PR, I'm inclined to reflect the wording of the product as-is. I agree that it's a confusing name though.

I think the "Platform Observability" effort from @andresrc might eliminate this name over time since we'll probably have things like "Logstash" dashboards that aren't specifically tied to any cluster uuid. At which point we might not need a name for it since it'll simply be the difference of having a cluster filter or not.

Reply:
Yes for this PR, makes sense. But I'm curious to understand what the intention of that name is. Andres's PO initiative is just a new name for what we've been working towards this whole time so it'll be up to us to make sure we define these things better in the future. 

I guess the real key is understanding why we need cluster_uuid to power the UI, ie is this a problem with the name or with the data. Definitely not something to worry about for this docs PR but hopefully these docs updates will help us identify these gaps and start closing them over time :)

Reply:
I see it as a problem with the current Stack Monitoring UI design.

The main entry point is a list of ES clusters, so you need to invent some stand-in when dealing with stack components that lack associated ES clusters (or at least lack the monitoring data).

Reply:
Great stuff @chrisronline ! I’ll work some of your wording into the terminology entry.